### PR TITLE
Mm 366 add one mixpanel event to app flow

### PR DESCRIPTION
**Pull request by noahsilencelab**

### CODE REVIEW

File: lib/repository/app_repository.dart [hunk 1]
```diff
@@ -32,9 +34,19 @@ class AppRepository extends DemoDecoratorComposite {
     if (backup != null) {
       return _pair(qrMessage, userId, backup);
     } else {
-      return _pair(qrMessage, userId) //
-          .then((_) => _sdk.startKeygen().value)
-          .then((keyshare) => _sdk.fetchRemoteBackup(keyshare.ethAddress).value);
+      return _pair(qrMessage, userId).then((_) async {
```
Comment (Granfalloner) [inline lib/repository/app_repository.dart]: Let's extract this handler into helper keygen method - high level pairing logic now is interspersed with lower-level tracking logic.

File: lib/third_party/analytics.dart [hunk 2]
```diff
@@ -15,7 +15,11 @@ enum SignInMethod { gmail, apple }
 
 enum PairingDeviceStatus { qr_scanned, success, failed }
 
-enum PairingDeviceType { repaired, recovered, start }
+enum PairingDeviceType { repaired, recovered, new_account }
```
Comment (Granfalloner) [inline lib/third_party/analytics.dart:18]: Is it intentional to have **`new_account`** in both `PairingDeviceType` and `DistributedKeyGenType`?

Comment (noahsilencelab) [reply lib/third_party/analytics.dart:18]: Yes, it is

File: lib/repository/app_repository.dart (hunk 1 above)
Comment (noahsilencelab) [reply lib/repository/app_repository.dart]: You mean sth like this:
```dart

    if (backup != null) {
      return _pair(qrMessage, userId, backup);
    } else {
      return _pair(qrMessage, userId).then((_) async {
        return await _keygen();
      }).then((keyshare) => _sdk.fetchRemoteBackup(keyshare.ethAddress).value);
    }
  }

  Future<Keyshare2> _keygen() async {
    try {
      _analyticManager.trackDistributedKeyGen(type: DistributedKeyGenType.new_account, status: DistributedKeyGenStatus.initiated);
      final keyshare = await _sdk.startKeygen().value;
      _analyticManager.trackDistributedKeyGen(
          type: DistributedKeyGenType.new_account, status: DistributedKeyGenStatus.success, publicKey: keyshare.ethAddress);
      return keyshare;
    } catch (error) {
      _analyticManager.trackDistributedKeyGen(
          type: DistributedKeyGenType.new_account, status: DistributedKeyGenStatus.failed, error: error.toString());
      rethrow;
    }
  }
  ```

Comment (Granfalloner) [reply lib/repository/app_repository.dart]: Yes, but do why do we need to use await before _keygen call? Can't we just call it like
```suggestion
return _pair(qrMessage, userId) //
  .then((_) => _keygen())
  .then((keyshare) => _sdk.fetchRemoteBackup(keyshare.ethAddress).value);```

Comment (noahsilencelab) [reply lib/repository/app_repository.dart]: We can't

**Review by Granfalloner — APPROVED** [review]

**Pull request merged**